core/src/main/scala/latis/input/ColumnarAdapter.scala

```
@@ -0,0 +1,39 @@
+package latis.input
+
+import latis.model.DataType
+
+/**
+ * Use column index properties to parse ascii tabular data.
+ * Map Variable names to zero-based column index(s).
+ * Each variable spec will be separated by ";" and
+ * column indices for multi-column variables will
+ * be separated by ",". e.g. columns="0,1,2;5;3;4"
+ */
+class ColumnarAdapter(
+  model: DataType,
+  config: ColumnarAdapter.Config = new ColumnarAdapter.Config()
+) extends TextAdapter(model, config) {
+
+  lazy val columnIndices: Vector[Array[Int]] = config.columns match {
+    case Some(s: String) => s.split(";").map(p => p.split(",").map(_.toInt)).toVector
+    case None => throw new RuntimeException("ColumnarAdapter requires 'columns' definition.")
+  }
+
+  override def extractValues(record: String): Vector[String] = {
+    //Note, trim record first to deal with leading white space.
+    val ss = splitAtDelim(record).filterNot(_.isEmpty)
+    //Ignore rows with fewer columns than those requested
+    if(ss.length <= columnIndices.flatten.max) Vector()
+    //append with " " for now since delimiter could be a regex
+    else columnIndices.map(is => is.map(ss(_)).mkString(" "))
+  }
+}
+
+object ColumnarAdapter {
+  /**
+   * Configuration specific to a ColumnarAdapter.
+   */
+  class Config(properties: (String, String)*) extends TextAdapter.Config(properties: _*) {
+    val columns: Option[String] = get("columns")
```

I think it is safe to say that the "columns" property is required. We could capture that in type by returning `String` and fail fast (via an exception, for now?) if it's not defined.